### ISSUE TITLE
Colon rule - apply_to_dictionaries: false is not respected

### ISSUE THREAD

**Issue by DG0BAB** [realm/SwiftLint#1901]
### New Issue Checklist

- ✅ Updated SwiftLint to the latest version
- ✅ I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

Since version 0.23.0 the "apply_to_dictionaries: false" for the colon rule  isn't respected anymore. In our Project we specify dictionaries like `[key : value]`. The bug only appears if value is of type Any. This worked fine until version 0.23.0 with "apply_to_dictionaries: false"

### Environment

* SwiftLint version 0.23.1
* Installation method SwiftLint.pkg

```yml
line_length:
  warning: 180
  ignores_comments: true

colon:
  apply_to_dictionaries: false

identifier_name:
  min_length:
    error: 3
  max_length:
    warning: 60
  excluded:     # excluded via string array
    - qa
    - QA
    - id
    - i
    - iD
  allowed_symbols:
    - _

vertical_whitespace:
    max_empty_lines: 5

function_body_length:
    warning: 80
    error: 150

disabled_rules:
    todo

excluded: # paths to ignore during linting. overridden by `included`.
  - Carthage
```

* Using Xcode version 9.1 beta 2 (Build (9B46)

```bash
echo "[String : Any]" | swiftlint lint --no-cache --use-stdin --enable-all-rules
Loading configuration from '.swiftlint.yml'
<nopath>:1:2: warning: Colon Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
```

```swift
// This triggers a violation:
let foo: [String : Any]

// This does not trigger a violation:
let foo: [String : Int]
```

```bash
echo "[String : Int]" | swiftlint lint --no-cache --use-stdin --enable-all-rules
Loading configuration from '.swiftlint.yml'
Done linting! Found 0 violations, 0 serious in 1 file.
```

**Comment by serges147** [comment]
+1

**Comment by marcelofabri** [comment]
`apply_to_dictionaries` is only verified for dictionary literals. In terms of language, literals and dictionary type declarations are different things. Feel free to submit a PR to improve this, but my initial guess is that it might be tricky to get it right.

**Comment by DG0BAB** [comment]
Yeah, you are right. Only dictionary literals are ignored and not dictionary declarations. But that changed with version 0.23.x. In former versions even dictionary declarations were ignored. So currently we are still using version 0.22 as a workaround.

**Comment by broadwaylamb** [comment]
+1

**Comment by marcelofabri** [comment]
PR are welcome 😉

**Comment by jakewelton** [comment]
+1

**Comment by gcrevell** [comment]
+1

**Comment by ArthurMaroulier** [comment]
+1

**Comment by guidomb** [comment]
@marcelofabri I've just tested this on `master` and the issue seems to solved. Based on the commit history from version `0.25.0` to `master` it seems that https://github.com/realm/SwiftLint/commit/dddb0f6cdd0f486e41257406e9cdc2d07f8f131c fixed it.

Should this bug be closed?
Do you have an ETA for the next release?

**Comment by marcelofabri** [comment]
@guidomb I think this is still valid. dddb0f6 is about another issue and I don't believe it would affect this one.

**Comment by guidomb** [comment]
@marcelofabri Yeap my bad! I was using the wrong build. I'd like to help fixing this issue. Any tips where to take a look?

I guess [this](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/ColonRule+Dictionary.swift#L31) method and probably [this](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/ColonRule+Dictionary.swift#L53) are good starting points.

**Comment by guidomb** [comment]
@marcelofabri [Here](https://github.com/realm/SwiftLint/pull/2067/files) is a first attempt of fixing this issue but some tests are failing I'd like to see if my patch makes sense

**Comment by vanics** [comment]
+1

**Comment by p4checo** [comment]
+1 😇
